geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneFunctionSecurityTest.java

```diff
@@ -105,366 +67,45 @@
-    Function thisFunction = dumpDirectoryFiles;
-    ResourcePermission thisMissingPermission = CLUSTER_MANAGE;
-
-    gfsh.executeAndAssertThat("execute function --id=" + thisFunction.getId())
-        .containsOutput("not authorized for " + thisMissingPermission.toString()).statusIsError();
-  }
-
-  @Test
-  @ConnectionConfiguration(user = "clusterManage", password = "clusterManage")
-  public void testInvalidPermissionsForDumpDirectoryFilesWithoutRegionParameter_withClusterManage() {
-    Function thisFunction = dumpDirectoryFiles;
-    ResourcePermission thisMissingPermission = DATA_READ;
-
-    gfsh.executeAndAssertThat("execute function --id=" + thisFunction.getId())
-        .containsOutput("not authorized for " + thisMissingPermission.toString()).statusIsError();
-  }
-
-  @Test
-  @ConnectionConfiguration(user = "noPermissions", password = "noPermissions")
-  public void testInvalidPermissionsForDumpDirectoryFilesWithRegionParameter_noPermission()
-      throws Exception {
-    Function thisFunction = dumpDirectoryFiles;
-
-    Predicate<String> notAuthForDataReadRegion =
-        s -> s.contains("not authorized for " + DATA_READ_REGION.toString());
-    Predicate<String> notAuthForClusterManage =
-        s -> s.contains("not authorized for " + CLUSTER_MANAGE.toString());
-    Predicate<String> notAuthForSomePermission =
-        s -> notAuthForDataReadRegion.test(s) || notAuthForClusterManage.test(s);
-
-    String output =
-        gfsh.execute("execute function --region=" + regionName + " --id=" + thisFunction.getId());
-
-    Condition<String> containsSomeAuthFailure =
-        new Condition<>(notAuthForSomePermission, "D:R or C:M:L auth failure", output);
-    assertThat(output).has(containsSomeAuthFailure);
-  }
-
-  @Test
-  @ConnectionConfiguration(user = "dataReadThis_test_region", password = "dataReadThis_test_region")
-  public void testInvalidPermissionsForDumpDirectoryFilesWithRegionParameter_withDataReadRegion() {
-    Function thisFunction = dumpDirectoryFiles;
-    ResourcePermission thisMissingPermission = CLUSTER_MANAGE;
-
-    gfsh.executeAndAssertThat(
-        "execute function  --region=" + regionName + " --id=" + thisFunction.getId())
-        .containsOutput("not authorized for " + thisMissingPermission.toString()).statusIsError();
-  }
 
   @Test
-  @ConnectionConfiguration(user = "clusterManage", password = "clusterManage")
-  public void testInvalidPermissionsForDumpDirectoryFilesWithRegionParameter_withClusterManage() {
-    Function thisFunction = dumpDirectoryFiles;
-    ResourcePermission thisMissingPermission = DATA_READ;
-
-    gfsh.executeAndAssertThat(
-        "execute function  --region=" + regionName + " --id=" + thisFunction.getId())
-        .containsOutput("not authorized for " + thisMissingPermission.toString()).statusIsError();
+  @ConnectionConfiguration(user = "user", password = "user")
+  public void functionRequireExpectedPermission() throws Exception {
```

Comment:
I understand that the testing of the security service itself is rather beyond the scope of this test class, but do we have any positive-tests for permissions with region-level specificity?  `ExecuteFunctionCommandSecurityTest` and `ExecuteFunctionCommandWitSecurityDUnitTest` appear to stop at the second tier, i.e., `clusterManage` and `dataRead`.

I think we need the positive-test coverage that these `testValid*` tests have, even if they belong in a different class / ticket.

Reply:
We have plenty of unit tests that tests implies to target level. I will add a few more here if needed.